Income 삭제

jiyongYoon opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
- Income 삭제 API 기능 구현
- Income 삭제 API 기능 Test코드 작성

**TO-BE**
- Exception 구현

### 테스트 
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [x] 테스트 코드
- [x] API 테스트

jiyongYoon committed:
FEAT /Income 삭제 API 기능 구현

jiyongYoon committed:
FEAT /Income 삭제 API 기능 Test코드 작성

rmsguqsla commented on server/src/main/java/com/rezero/inandout/income/controller/IncomeController.java:
```diff
@@ -61,4 +62,14 @@ public ResponseEntity<?> getIncomeListAndDetailCategoryList(Principal principal,
+
+    @DeleteMapping
+    public ResponseEntity<?> deleteIncome(Principal principal,
+                             @RequestBody List<Long> deleteIdList) {
```
Long말고 클래스 하나를 추가해줘야 할 것 같습니다!!
그래야 Json에서 매핑이 가능할 것 같네용^^

rmsguqsla commented on server/src/main/java/com/rezero/inandout/income/service/IncomeServiceImpl.java:
```diff
@@ -119,6 +119,14 @@ public void updateIncome(String email, List<IncomeInput> incomeInputList) {
 
     }
 
+    @Override
+    public void deleteIncome(String email, List<Long> deleteIdList) {
+        memberRepository.findByEmail(email)
+            .orElseThrow(() -> new RuntimeException("없는 맴버입니다."));
+
+        incomeRepository.deleteAllByIdInBatch(deleteIdList);
+    }
+
```
저는 memberId와 incomeId가 같이 검증해야할 것 같다고 생각해서
deleteIdList에 대해서 for문 돌리고 그 안에 deleteByIncomeIdAndMember이렇게 했네요
혹시 제 방식에 대해서 어떤 의견이신지 궁금합니다!!

jiyongYoon replied:
오 그거 좋은 방법이네요, 저도 추가로 더 검증할게 없나 고민했었는데,,, 
또 수정해야겠어요^^

rmsguqsla replied:
근데 deleteAllByIdInBatch가 성능면에선 뛰어나기 때문에 저도 고민한 부분이긴하네요

jiyongYoon replied:
아 생각해보니 InBatch는 살리고, for문 돌리면서 List<Long> 으로 받기 전에 id검사만 한번 하면 어떨까요?

rmsguqsla replied:
화면공유로 설명가능하실까요??

jiyongYoon committed:
FEAT /Income 삭제 API 기능 Test코드 수정

rmsguqsla reviewed (commented):
고생하셨습니다~

jiyongYoon committed:
FEAT /Income 삭제 API 기능 구현 (+Input클래스 추가)

rmsguqsla commented:
커뮤니케이션이 되고 코드가 발전하니까 좋네요^^ 고생하셨습니다

jiyongYoon committed:
FEAT /Income 삭제 API 기능 수정 및 테스트코드 수정
- Income의 memberId와 현재 로그인한 memberId가 동일한지 검증 추가.
- update 메서드 및 테스트코드도 동일한 이유로 수정함.
  -> 해당 수입내역의 유저가 아닌 테스트코드 추가함.
- repository 문법오류 수정.

goraneee commented on server/src/test/java/com/rezero/inandout/income/controller/IncomeControllerTest.java:
```diff
@@ -189,11 +190,15 @@ void deleteIncome() throws Exception {
-        String deleteIdListJson = objectMapper.writeValueAsString(deleteIdList);
+        String deleteIdListJson = objectMapper.writeValueAsString(deleteIncomeInputList);
 
         User user = new User(member.getEmail(), member.getPassword(), AuthorityUtils.createAuthorityList("ROLE_USER"));
```
저 User  객체 첨 본 거 같아요~
원래 암호화 안 하고 이런 식으로 쓰는 건가요?

rmsguqsla replied:
세션방식이면 테스트를 저렇게 해야하는 것 같아요

goraneee commented:
확인완료했습니당 ~
